lib/grinder_files.dart

```diff
@@ -98,12 +100,210 @@ class FileSet {
+  }
+
+  /**
+   * Returns the containing [Path]. Returns a non-null value even if this is a
+   * root directory.
+   *
+   * See [FileSystemEntity.parent].
+   */
+  Path get parent {
+    int index = _path.lastIndexOf(_sep);
+
+    // Do string manipulation if there are path separators; otherwise, use the
+    // file system entity information.
+    if (index == 0 || index == -1) {
+      FileSystemEntity e = entity;
+      return e == null ? null : new Path(e.parent);
+    } else {
+      return new Path(_path.substring(0, index));
+    }
+  }
+
+//  /**
+//   * Returns the abolute version of this Path.
+//   */
+//  Path get absolute {
+//    // TODO:
+//  }
+
+  bool get isDirectory => FileSystemEntity.isDirectorySync(_path);
+  bool get isFile => FileSystemEntity.isFileSync(_path);
+  bool get isLink => FileSystemEntity.isLinkSync(_path);
+
+  /**
+   * Assume the current file system entity is a [File] and return it as such.
+   * You would call this instead of [entity] when the file system entity does
+   * not yet exist.
+   */
+  File get asFile => new File(path);
+
+  /**
+   * Assume the current file system entity is a [Directory] and return it as such.
+   * You would call this instead of [entity] when the file system entity does
+   * not yet exist.
+   */
+  Directory get asDirectory => new Directory(path);
+
+  /**
+   * Assume the current file system entity is a [Link] and return it as such.
+   * You would call this instead of [entity] when the file system entity does
+   * not yet exist.
+   */
+  Link get asLink => new Link(path);
+
+  /**
+   * Copy the the entity to the given destination. Return the newly created
+   * [Path].
+   */
+  Path copy(Path destDir) {
+    _copyImpl(entity, destDir.asDirectory);
+    return new Path(destDir).join(name);
+  }
+
+  /**
+   * Delete the entity at the path.
+   */
+  void delete() => _deleteImpl(entity);
+
+  /**
+   * Synchronously create the file. See also [File.create].
+   *
+   * If [recursive] is false, the default, the file is created
+   * only if all directories in the path exist. If [recursive] is true, all
+   * non-existing path components are created.
+   */
+  void createFile({bool recursive: false}) {
+    asFile.createSync(recursive: recursive);
+  }
+
+  /**
+   * Synchronously create the directory. See also [Directory.create].
+   *
+   * If [recursive] is false, the default, the file is created
+   * only if all directories in the path exist. If [recursive] is true, all
+   * non-existing path components are created.
+   */
+  void createDirectory({bool recursive: false}) {
+    asDirectory.createSync(recursive: recursive);
+  }
+
+  /**
+   * Synchronously create the link. See also [Link.create].
+   *
+   * If [recursive] is false, the default, the file is created
+   * only if all directories in the path exist. If [recursive] is true, all
+   * non-existing path components are created.
+   */
+  void createLink(String target, {bool recursive: false}) {
```

Comment:
`copy` takes a `Path`, while this takes a `String`.  Can we settle on one or allow multiple?  Also, what about `FileSystemEntity`?